here-naksha-app-service/src/test/java/com/here/naksha/app/service/util/ApiUtilTest.java

```diff
@@ -18,102 +18,147 @@
-    assertEquals("ten", innerOpList.get(0).getPropertyRef().getTagName());
-    assertEquals(POpType.EXISTS, innerOpList.get(1).op());
-    assertEquals("eleven", innerOpList.get(1).getPropertyRef().getTagName());
+    assertThatOperation(orList.get(innerOpsInd++))
+        .hasType(OpType.AND)
+        .hasChildrenThat(
+            tenth -> tenth.existsWithTagName("ten"), eleventh -> eleventh.existsWithTagName("eleven"));
+
     // validate 7th operation uses OR
-    crtOp = orList.get(i++);
-    assertEquals(POpType.OR, crtOp.op());
-    innerOpList = crtOp.children();
-    assertNotNull(innerOpList, "Expected multiple operations");
-    assertEquals(2, innerOpList.size());
-    assertEquals(POpType.EXISTS, innerOpList.get(0).op());
-    assertEquals("twelve", innerOpList.get(0).getPropertyRef().getTagName());
-    assertEquals(POpType.EXISTS, innerOpList.get(1).op());
-    assertEquals("thirteen", innerOpList.get(1).getPropertyRef().getTagName());
+    assertThatOperation(orList.get(innerOpsInd++))
+        .hasType(OpType.OR)
+        .hasChildrenThat(
+            twelfth -> twelfth.existsWithTagName("twelve"),
+            thirteenth -> thirteenth.existsWithTagName("thirteen"));
+
     // validate 8th operation uses EXISTS
-    crtOp = orList.get(i++);
-    assertEquals(POpType.EXISTS, crtOp.op());
-    assertEquals("fourteen", crtOp.getPropertyRef().getTagName());
+    assertThatOperation(orList.get(innerOpsInd)).existsWithTagName("fourteen");
+  }
+
+  @ParameterizedTest
+  @MethodSource("simpleTagsSample")
+  void shouldParseSimpleTags(String queryString, Consumer<POpAssertion> assertion) {
+    QueryParameterList queryParameters = new QueryParameterList(queryString);
+    POp op = ApiUtil.buildOperationForTagsQueryParam(queryParameters);
+    assertion.accept(new POpAssertion(op));
+  }
+
+  private static Stream<Arguments> simpleTagsSample() {
+    return Stream.of(
+        tagQuerySpec("tags=x", op -> op.existsWithTagName("x"), "only 'x'"),
+        tagQuerySpec(
+            "tags=this,that",
+            op -> op.hasType(OpType.OR)
+                .hasChildrenThat(
+                    child -> child.existsWithTagName("this"),
+                    child -> child.existsWithTagName("that")),
+            "'this' or 'that'"),
+        tagQuerySpec(
+            "tags=foo+bar",
+            op -> op.hasType(OpType.AND)
+                .hasChildrenThat(
+                    child -> child.existsWithTagName("foo"),
+                    child -> child.existsWithTagName("bar")),
+            "'foo' and 'bar'"));
+  }
+
+  private static Arguments tagQuerySpec(String query, Consumer<POpAssertion> assertion, String assertionDesc) {
+    return arguments(query, named(assertionDesc, assertion));
+  }
+
+  private static POpAssertion assertThatOperation(POp subject) {
+    assertNotNull(subject);
+    return new POpAssertion(subject);
+  }
+
+  private record POpAssertion(POp subject) {
```

Comment:
POpAssertion looks to be useful at other such places in future. Can create a separate class.